omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/DiagnosisController2_2Test.java

```diff
@@ -0,0 +1,158 @@
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import static org.openmrs.ConditionVerificationStatus.CONFIRMED;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.DiagnosisService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_2;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link DiagnosisController2_2}.
+ */
+public class DiagnosisController2_2Test extends MainResourceControllerTest {
+
+	private DiagnosisService diagnosisService;
+
+	private Encounter encounter;
+
+	@Before
+	public void before() throws Exception {
+		executeDataSet(RestTestConstants2_2.DIAGNOSIS_TEST_DATA_XML);
+		this.diagnosisService = Context.getDiagnosisService();
+		this.encounter = new Encounter();
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getURI()
+	 */
+	@Override
+	public String getURI() {
+		return RestConstants.VERSION_1 + "/patientdiagnoses";
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_2.DIAGNOSIS_UUID;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return diagnosisService.getPrimaryDiagnoses(this.encounter).size();
+	}
+
+	@Test
+	public void shouldCreateADiagnosis() throws Exception {
+		long originalCount = getAllCount();
+
+		SimpleObject diagnosisSource = new SimpleObject();
+		diagnosisSource.add("diagnosisId", 1);
+		diagnosisSource.add("encounter", new Encounter());
+		diagnosisSource.add("diagnosis", new CodedOrFreeText());
+		diagnosisSource.add("certainty", CONFIRMED);
+		diagnosisSource.add("rank", 1);
+
+		String json = new ObjectMapper().writeValueAsString(diagnosisSource);
+
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+
+		SimpleObject newDiagnosisSource = deserialize(handle(req));
+
+		Assert.assertNotNull(PropertyUtils.getProperty(newDiagnosisSource, "uuid"));
+
+		Diagnosis diagnosis = diagnosisService.getDiagnosisByUuid(getUuid());
+
+		Assert.assertEquals(diagnosis.getId(), diagnosisSource.get("diagnosisId"));
```

Comment:
Shouldn't you be using newDiagnosisSource instead of diagnosisSource?

Reply:
That was an error. Rectifying this.